pkg/types/baremetal/platform.go

```diff
@@ -71,7 +71,7 @@ type Platform struct {
 	ProvisioningDHCPRange string `json:"provisioningDHCPRange,omitempty"`
 
 	// Hosts is the information needed to create the objects in Ironic.
-	Hosts []*Host `json:"hosts" validate:"omitempty,dive"`
+	Hosts []*Host `json:"hosts" validate:"hostscount,omitempty,dive"`
```

Comment:
I'm a little bit hesitant to say we should use annotations for all validations.

I don't this isn't reusable for anything else, unlike other validations. TBH, I think it makes sense for this just to be a normal validation with a very simple check in the body of the ValidatePlatform function.  I would find that code easier to read.

What do you think?

Reply:
Also: what does 'dive' mean?

Reply:
Well, I also had some doubts about the reusability of this rule and agreed that very likely it will not be reusable elsewhere. I followed such approach for a matter of maintenance/readability (it is sufficient to look at the struct tags to understand all the validations applied to a given field) and because the boilerplate code required to add a new validation rule was essentially the same in this case (`types.InstallConfig` must always be passed as a contextual info to apply the cross-struct validation). I can revert it back to a normal validation to see how it looks like.